dmapiclient/data.py

```diff
@@ -239,10 +239,19 @@ def agree_framework_variation(self, supplier_id, framework_slug, variation_slug,
             user=user,
         )
 
-    def find_framework_suppliers(self, framework_slug, agreement_returned=None):
+    def find_framework_suppliers(self, framework_slug, agreement_returned=None, statuses=None):
+        '''
+        :param agreement_returned: A boolean value that allows filtering by suppliers who have or have not
+                                   returned their framework agreement. If 'agreement_returned' is set then
+                                   any value for 'statuses' will be ignored.
+        :param statuses: A comma-separated list of the statuses of framework agreements that should be returned.
+                         Valid statuses are: signed, on-hold, approved and countersigned.
+        '''
         params = {}
         if agreement_returned is not None:
             params['agreement_returned'] = bool(agreement_returned)
+        elif statuses is not None:
+            params['status'] = statuses
```

Comment:
I'd consider this `elif` to be inserting too much business-logic into the apiclient. It knows too much about the nature of these `SupplierFrameworks` - that you won't get any valid statuses if you're not setting `agreement_returned`.

In fact, I'd argue that the behaviour is wrong - you should be able to filter by `status` without filtering by `agreement_returned`. The latter would just be implicit.